external-crates/move/crates/move-analyzer/tests/ide_testsuite.rs

```diff
@@ -436,45 +446,119 @@ fn completion_test(
 // Test Suite Runner Code
 //**************************************************************************************************
 
-fn initial_symbols(
-    project: String,
-) -> datatest_stable::Result<(PathBuf, CompiledPkgInfo, Symbols)> {
-    let base_path = PathBuf::from(env!("CARGO_MANIFEST_DIR"));
-    let mut project_path = base_path.clone();
-    project_path.push(project);
-
-    let ide_files_root: VfsPath = MemoryFS::new().into();
-    let pkg_deps = Arc::new(Mutex::new(CachedPackages::new()));
-
-    // Similarly to `get_symbols`, we retry to exercise caching
-    let mut should_retry = true;
-    loop {
-        let (compiled_pkg_info_opt, _) = get_compiled_pkg(
-            pkg_deps.clone(),
-            ide_files_root.clone(),
-            project_path.as_path(),
-            LintLevel::None,
-            BTreeMap::new(),
-            Some(Flavor::Sui),
-        )?;
-        let compiled_pkg_info = compiled_pkg_info_opt.ok_or("PACKAGE COMPILATION FAILED")?;
-        let symbols = compute_symbols(pkg_deps.clone(), compiled_pkg_info.clone(), None);
-        if !should_retry {
-            return Ok((project_path, compiled_pkg_info, symbols));
+/// Compute symbols with optional file modifications to trigger incremental compilation.
+///
+/// When `file_modifications` is None, performs full compilation.
+/// When `file_modifications` is Some, writes modified content to VFS overlay,
+/// which triggers incremental compilation by causing file hash mismatches.
+///
+/// Returns both CompiledPkgInfo and Symbols for test suites that need both.
+fn test_symbols_with_optional_modifications(
```

Comment:
This is a somewhat unrelated change that aims of testing caching subsystem to a larger degree - actually modify the file (in the virtual file system) to trigger incremental compilation/symbolication